### ISSUE TITLE
Failing to connect can cause all subsequent attempts to fail

### ISSUE THREAD

**Issue by Ijwu** [Ijwu/Archipelago.HollowKnight#7]
Seems to happen when socket was unexpectedly closed.

**Comment by dewiniaid** [comment]
I can no longer reproduce this under either of these two test cases:

- Attempt to connect to the wrong port first, then connect to the correct one
- Disconnect from AP mid-game, then reload the save.

I'm assuming the 0.0.3 revamp fixed this, and we can reopen if it turns out I'm wrong.

**Comment by good-praxis** [comment]
I can't reproduce this on a clean save, but I ran into an failing-all-subsequent-attempts issue on 0.1.0.1. that loosely fits the description:

I initially connected to the wrong port, then connected to the correct one, managed a 2h session. Two weeks later I tried to reconnect to the server before turning it on, then tried again with it on leading to the "connected" message in the log, followed by a disconnection before the save loads. I verified via the tracker that the save is at the state it should be at, but I was unable to reproduce it on a clean server
